Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -255,6 +255,152 @@ function Get-CertificateFromDomainController {
     return $SSHOutput
 }
 
+<#
+    .Synopsis
+     Recommended: Debug functionality of all OpenLDAP and Active Directory Identity Sources for use with vCenter Server Single Sign-On.
+
+    .Example
+    # Debug the OpenLDAP and Active Directory Identity Sources configured in vCenter
+    Debug-LDAPSIdentitySources
+#>
+function Debug-LDAPSIdentitySources {
+    [AVSAttribute(2, UpdatesSDDC = $false)]
+
+    Write-Host "*"
+    Write-Host "* LDAP Identity Source Diagnostic Test Tool (ldapcheck)"
+    Write-Host "* Executed at $((Get-Date).ToUniversalTime())"
+    Write-Host "*"
+
+    $sources = Get-IdentitySource -External -ErrorAction Stop
+    $sources | ForEach-Object {
+        if(($_.Type -eq "OpenLdap") -or ($_.Type -eq "ActiveDirectory")) {
+
+            Write-Host "* -------------------------------------------------------------"
```

Comment:
replace Write-Host with Write-Information Write-Warning Write-Error

Reply:
Will that have any effect, in the context on an internal run-command? As far an consistency, the other already-present functions appear to use a mix of either just Write-Host or some use all the variations.

Reply:
Write-Host is used to convey messages about script running, the outcomes (eg success/failure) are passed via the above to the info/warning/error stream via the above